### PR TITLE
radare2: Disable compiler flag unavailable outside OSX

### PR DESCRIPTION
**Pull request by dholm**

### CODE REVIEW

**Comment by dholm** [comment]
Will fail since the flag is custom to Clang on OSX.

https://gist.github.com/dholm/0b424c2d6e287caf9aa26b4483d7f335

**Comment by dholm** [comment]
@sjackman The failures seems to come from brew not being able to download any files?

**Comment by dholm** [comment]
It still fails to download valabind.  I ran the curl command locally and removed `--silent` and looks like the certificate verification fails for `www.radare.org` (at least on my computer).

**Comment by sjackman** [comment]
```
==> Downloading https://www.radare.org/get/valabind-0.10.0.tar.gz
/usr/bin/curl --remote-time --location --user-agent Linuxbrew/1.2.0 (Linux; x86_64 Ubuntu 14.04.5 LTS) curl/7.35.0 --fail --silent https://www.radare.org/get/valabind-0.10.0.tar.gz -C 0 -o /home/linuxbrew/.cache/Homebrew/valabind-0.10.0.tar.gz.incomplete
Error: Failed to download resource "valabind"
Download failed: https://www.radare.org/get/valabind-0.10.0.tar.gz
```
https://travis-ci.org/Linuxbrew/homebrew-core/builds/237828973#L1135

**Comment by sjackman** [comment]
`curl https://www.radare.org/get/valabind-0.10.0.tar.gz` works for me on macOS 10.11.6. It fails for me on Linuxbrew:
```
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
```

**Comment by sjackman** [comment]
The Linuxbrew OpenSSL certs are current at version 2017-01-18 from http://curl.haxx.se/docs/caextract.html

File: Formula/radare2.rb [hunk 1]
```diff
@@ -104,8 +104,10 @@ def install
       lua_version = Utils.popen_read("lua -v 2>&1").match(/%r{5\.\d}/)
 
       # Lazily bind to Python.
```
Comment (sjackman) [inline Formula/radare2.rb]: Please move this comment inside the `if` block.

File: Formula/radare2.rb [hunk 2]
```diff
@@ -59,7 +59,7 @@ class Radare2 < Formula
     end
   end
 
-  option "with-code-signing", "Codesign executables to provide unprivileged process attachment"
+  option "with-code-signing", "Codesign executables to provide unprivileged process attachment" if OS.mac?
```
Comment (sjackman) [inline Formula/radare2.rb]: Please omit this change. Granted the option is not valid on Linux, but it doesn't hurt anything (unless the user tries to enable it).

**Comment by sjackman** [comment]
My suspicion is that the SSL certificate of `www.radare.org` is somehow misconfigured. The workaround is here: https://github.com/Linuxbrew/brew/wiki/FAQ#why-does-curl-fail

**Comment by sjackman** [comment]
I'll merge this PR without updating the bottles once the requested changes are made.

File: Formula/radare2.rb (hunk 2 above)
Comment (dholm) [inline Formula/radare2.rb]: Done

File: Formula/radare2.rb (hunk 1 above)
Comment (dholm) [inline Formula/radare2.rb]: Done

**Comment by sjackman** [comment]
I'll merge this PR this afternoon once PR #2822 is merged.

**Comment by sjackman** [comment]
Merged. Thanks again for your contribution to Linuxbrew, David!

**Pull request merged**